DSC/dsc.py

```diff
@@ -199,24 +206,102 @@ def enable():
                                               isSuccess=False,
                                               message="(03107)Failed to apply meta MOF configuration through Pull Mode")
                 hutil.do_exit(1, 'Enable', 'error', '1', 'Enable failed. ' + current_config)
-                
+        
+        send_heart_beat_msg_to_agent_service()
+        
         agent_id = get_nodeid(nodeid_path)
         vm_uuid = get_vmuuid()
         if vm_uuid is not None and agent_id is not None:
             status_filepath = get_statusfile_path()
             update_statusfile(status_filepath, agent_id, vm_uuid)
-	    sys.exit(0)
+            sys.exit(0)
               
         hutil.do_exit(0, 'Enable', 'success', '0', 'Enable Succeeded')
     except Exception as e:
         waagent.AddExtensionEvent(name=ExtensionShortName, op='EnableInProgress', isSuccess=True, message="Enable failed with the error: {0}, stacktrace: {1} ".format(str(e), traceback.format_exc()))
         hutil.error('Failed to enable the extension with error: %s, stack trace: %s' %(str(e), traceback.format_exc()))
         hutil.do_exit(1, 'Enable', 'error', '1', 'Enable failed: {0}'.format(e))
+
+def send_heart_beat_msg_to_agent_service():
+    try:
+        waagent.AddExtensionEvent(name=ExtensionShortName, op='HeartBeatInProgress', isSuccess=True, message="In send_heart_beat_msg_to_agent_service method")
+        code,output = run_cmd("python /opt/microsoft/dsc/Scripts/GetDscLocalConfigurationManager.py")
+        if code == 0 and "RefreshMode=Pull" in output:
+            waagent.AddExtensionEvent(name=ExtensionShortName, op='HeartBeatInProgress', isSuccess=True, message="sends heartbeat message in pullmode")
+            m = re.search("ServerURL=([^\n]+)", output)
+            if not m:
+                return
+            registration_url = m.group(1)
+            agent_id = get_nodeid(nodeid_path)
+            node_extended_properties_url = registration_url + "/Nodes(AgentId='" + agent_id + "')/ExtendedProperties"
+            waagent.AddExtensionEvent(name=ExtensionShortName, op='HeartBeatInProgress', isSuccess=True, message="Url is " + node_extended_properties_url)
+            headers = {'Content-Type': "application/json; charset=utf-8", 'Accept': "application/json", "ProtocolVersion" : "2.0"}
+            data = construct_node_extension_properties(output)
+            
+            request = urllib2httpclient.Urllib2HttpClient("/etc/opt/omi/ssl/oaas.crt", "/etc/opt/omi/ssl/oaas.key")
```

Comment:
Could you please make the paths as variables? Would help to change in one location.